src/pages/smart-contracts/stacks-api-client-tutorial.md

```diff
@@ -0,0 +1,727 @@
+```
+
+### 2. Step: Fetch the map entry
+
+With the serialized key, add the following lines to the fetchNewestRegistry callback
+
+```js
+// fetch public url and name
+const mapEntryResponseRaw = await smartContractsApi.getContractDataMapEntryRaw({
+  stacksAddress: CONTRACT_ADDRESS,
+  contractName: CONTRACT_NAME,
+  mapName: 'registry',
+  key,
+  proof: 0,
+});
+const mapEntryResponse = await mapEntryResponseRaw.raw.json();
+console.log({ mapEntryResponse });
+```
+
+~> Bug: Due to an open with the client library, `getContractDataMapEntryRaw` needs to be used instead of `getContractDataMapEntry`. Using `raw.json()` gives the same result as `getContractDataMapEntry` should do.
+
+### 3. Step: Deserialized data map entry
+
+The response object has a data property that is a hex representation of the map entry. The data map entry is, again, always a tuple. The `TupleCV` class has a property `data`. It is a dictionary of clarity values.
+
+To cover the case of an invalid key, the result is wrapped in an optional object. Hence, the type of the result can be `OptionalNone` or `OptionalSome`. The latter type as a property `value` that contains the tuple value.
+
+Therefore the deserialization of the result looks like this:
+
+```js
+const optionalMapEntry = deserializeCV(Buffer.from(mapEntryResponse.data.substr(2), 'hex'));
+console.log({ optionalMapEntry });
+if (optionalMapEntry.type === ClarityType.OptionalSome) {
+  const mapEntryCV = optionalMapEntry.value;
+  const registryData = mapEntryCV.data;
+}
+```
+
+Add this, together with the update of the react state variable to the "fetch newest registry" part of the callback after the log statement:
+
+```js
+if (mapEntryResponse) {
+  const optionalMapEntry = deserializeCV(Buffer.from(mapEntryResponse.data.substr(2), 'hex'));
+  console.log({ optionalMapEntry });
+  if (optionalMapEntry.type === ClarityType.OptionalSome) {
+    const mapEntryCV = optionalMapEntry.value;
+    const registryData = mapEntryCV.data;
+    const name = cvToString(registryData.name);
+    const url = cvToString(registryData.url);
+
+    setNewestRegistation({
+      name,
+      url,
+      lastId: lastId.toString(),
+      owner,
+    });
+  }
+}
+```
+
+Now, all data are stored in `newestRegistration`.
+
+### 4. Step: Display url and name of registry entry
+
+Update the UI to display the url and name of the newest registry entry. Note, that `cvToString` places buffer values in quotes. Remove the quotes around the url using `substr`.
+
+```jsx
+{
+  newestRegistration && (
+    <Text fontWeight="500" display="block" mb={0} fontSize={2}>
+      <a href={newestRegistration.url.substr(1, newestRegistration.url.length - 2)}>
+        Newest Registration
+      </a>{' '}
+      by user {newestRegistration.name} using address {newestRegistration.owner} (registry id:{' '}
+      {newestRegistration.lastId})
+    </Text>
+  );
+}
+```
+
+This will show the newest registration with a link to the public to-do list.
+
+## Summary
```

Comment:
I wouldn't make a new chapter and I'd copy&paste the objectives from the intro (in this tutorial, you will..). With the objectives, you can re-iterate and make sure the reader puts the learnings into context